Create release communication guidelines (#304)
kyma-project/community: guidelines/releases/release-communication.md

- Inform Kyma teams about release deadlines on Slack channels at least one week before the first release candidate (RC1), and pin the message so that it is easy to find. 

- Prepare the environment to test a release candidate. Share access to the testing environment with Kyma teams so that they can test their components before merging the RC.

- Create an excel sheet where all the manual tests are defined and ensure that all tests are executed by responsible teams.

- Inform Kyma teams about deadlines for cherry-picking changes to the release branch.

- Inform Kyma teams about all the release-related pull requests that require approvals.

- Notify Kyma teams about failing tests to fix them as quickly as possible. Kyma teams responsible for failing tests should communicate with a Release Master about the fixing progress.

## Communication channels

The Release Master communicates the release progress simultaneously on these two internal Slack channels: 

- `c4core-xf-team` - the main channel to communicate with the Kyma teams

> - `c4core-kyma-scrum-masters` - a channel used to communicate the release progress to Scrum Masters

Reviewer: Those are private channels. I think we shouldn't mention them here.

Author: I deleted the information about them

# Release communication

This document describes a release process communication.

## Communication channels

The communication Kyma team should be conducted simultaneously on the two important channels: 

- `c4core-xf-team` - main channel to communicate about the release progress. 
- `c4core-kyma-scrum-masters` - a channel to escalate actions in case when some team lingers with their job. In that case you can also write directly to the proper Scrum Master.

The Kyma community should be notified about the release on the external channel. Our official channel for external announcements in `#general` on the `kyma-community` slack workspace.

## Communication persons

The Kyma team persons which are responsible for the releasing:

- [Jose Cortina](https://github.com/jose-cortina)

> ## Release master responsibilities

Reviewer 2: I'm not sure whether this section should be mentioned in this document. Please check whether it's not already described in the `release-process.md` or the `release-strategy.md`. Maybe these documents are better for info described in this section?

At the beginning of the release the releasing team should pick a release master which will manage the release process. 

The release master is obliged to do the following things:

> - Inform the Kyma team about release deadlines about one week before RC1. The release important dates should be pinned in all channels mentioned above. The Kyma team must know the planned date of the release to don't exceed them. The release master must watch if all Kyma teams respect the deadlines. In case not, he must escalate that to a proper channel or person.

Reviewer 2: ```suggestion
- Inform Kyma teams about release deadlines on Slack channels at least one week before RC1, and pin the message so that it is easy to find. 
```

> - Prepare the environment to test the release candidates at least 3 days before merging it. All kyma teams should use the prepared cluster to test their components. 

Reviewer 2: ```suggestion
- Prepare the environment to test a release candidate. Share access to the testing environment with Kyma teams so that they can test their components before merging the RC.
```

> - Create a excel sheet where all the manual tests are defined and he must take care that all of them are executed by the teams responsible for them. If some team doesnt executed their manual tests, that case must be escalated as well.

Reviewer 2: ```suggestion
- Create an excel sheet where all the manual tests are defined and ensure that all tests are executed by responsible teams.
```

> - Inform the Kyma teams about deadlines for cherry-picking to the release candidates.

Reviewer 2: ```suggestion
- Inform Kyma teams about deadlines for cherry-picking changes to the release branch.
```

> - Inform the Kyma teams about all the release related pull requests which requires approvals in order to merge them as quickly as possible.

Reviewer 2: ```suggestion
- Inform Kyma teams about all the release-related pull requests that require approvals.
```

> - Inform the Kyma teams which tests are failing to fix them as quickly as possible. It should be escalated if there is any risk that some test won't be fixed which will affect the release. The Kyma teams which are responsible for failing test should communicate with release master about fixing progress.

Reviewer 2: ```suggestion
- Notify Kyma teams about failing tests to fix them as quickly as possible. Kyma teams responsible for failing tests should communicate with a Release Master about the fixing progress.
```

> ## Communication rules

Reviewer 2: I'd remove this whole section as it doesn't bring any relevant info.

The messages about the release sent to the Kyma slack channels should be written transparently. They should inform about each aspect of the next release step.

It's recommended to write updates about releases using the multiline sentences which won't be missed among other messages in the channel.

When the updates about the release are important, they should be pinned to the channel so they will be more visible.

> ## Scrum of scrums meetings

Reviewer 2: ```suggestion
## Scrum of Scrums meetings
```

> Scrum of Scrums is the internal meeting for the Kyma Scrum Masters.

Reviewer 2: ```suggestion
A Release Master should cooperate with scrum masters regarding the release process communication. Either the Release Master or a scrum master dedicated to the releasing team must inform other scrum masters on the release progress during the biweekly Scrum of Scrums (SoS) meetings.
```

> The Scrum Master which team is responsible for the release should give a status about the release progress at the Scrum of Scrums meetings, so all of the Scrum Masters will know if there are some obstacles for the release.

Reviewer 2: remove

> The Scrum Master is also responsible for updating a Scrum of Scrums MagicBox under the Wiki page for the adequate release.

Reviewer 2: ```suggestion
A dedicated scrum master is also responsible for updating the SoS MagicBox under the Wiki page for a given release. The MagicBox must contain all the information and deadlines that affect all Kyma teams.
```
